openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -179,12 +181,17 @@ def get_eligibilities_for_user(username):
     Returns: list
 
     """
+    eligibilities = CreditEligibility.get_user_eligibilities(username)
+
+    if course_key:
+        eligibilities = eligibilities.filter(course_key=course_key)
+
     return [
         {
             "course_key": eligibility.course.course_key,
             "deadline": eligibility.deadline,
         }
-        for eligibility in CreditEligibility.get_user_eligibilities(username)
+        for eligibility in eligibilities
     ]
```

Comment:
@tasawernawaz Good question!

I would prefer that we add a separate API call just for the credit eligibility information, since deadlines are a property of the eligibility (not the provider).

How are you calling these APIs?  Is it a cross-domain AJAX call or a server-to-server API call?  If it's cross-domain AJAX, we should be able to execute both API calls in parallel to load different parts of the DOM, so it shouldn't have too big an impact on user-perceived performance.